Dart/src/com/jetbrains/lang/dart/ide/formatter/DartBlock.java

```diff
@@ -70,11 +77,41 @@ public Spacing getSpacing(Block child1, @NotNull Block child2) {
       return EMPTY;
     }
     final ArrayList<Block> tlChildren = new ArrayList<Block>();
-    for (ASTNode childNode = getNode().getFirstChildNode(); childNode != null; childNode = childNode.getTreeNext()) {
-      if (FormatterUtil.containsWhiteSpacesOnly(childNode)) continue;
-      final DartBlock childBlock = new DartBlock(childNode, createChildWrap(childNode), createChildAlignment(childNode), mySettings);
-      childBlock.setParent(this);
-      tlChildren.add(childBlock);
+    ASTNode parent = getNode();
+    if (parent.getElementType() == VALUE_EXPRESSION) {
```

Comment:
I'm afraid I didn't get what this code do. Isn't DartBlock tree good enough after your changes in parser? How this code modifies DartBlock tree compared to what was here before?